### DISCUSSION TITLE
all-day event shown at midnight (only)in android: icsx5s' fault?

### DISCUSSION THREAD

**Discussion by azrdev** [bitfireAT/icsx5#59] [Q&A] [answered]
I use icsx5 for syncing the calendar of [grocy](https://github.com/grocy/grocy) (v3.3.1), which contains "best before" dates of my household inventory. I also sync this .ics with Thunderbird/Lightning, where an example vevent can be copied like this:

~~~ical
BEGIN:VCALENDAR
PRODID:-//Mozilla.org/NONSGML Mozilla Calendar V1.1//EN
VERSION:2.0
BEGIN:VEVENT
DTSTAMP:20220904T121518Z
UID:5aef686d556328afad75d60a50d2c7fb
SUMMARY:Produkt fällig: Haferflocken
DTSTART;VALUE=DATE:20221202
END:VEVENT
END:VCALENDAR
~~~

Obviously this is an all-day event, and it renders as such.

In android, however, *the grocy events are all shown as timed events*, starting at 0:00 and having 0 duration, effectively hiding them in the "week" view I use by default.

Android is version 11, i.e. "LineageOS for MicroG" version 18.1 -- I see the problem with Etar from f-droid (version ) and the LineageOS builtin calendar v (a version of Etar).
When I share above calendar entry from the calendar, I get the following .ics from Etar:

~~~ical
BEGIN:VCALENDAR
VERSION:2.0
CALSCALE:GREGORIAN
METHOD:REQUEST
PRODID:-//Etar//ws.xsoh.etar
BEGIN:VEVENT
UID:1f468f4e-29c1-4f56-a11e-2aeabde9abf7@ws.xsoh.etar
DTSTAMP:20220904T122014Z
SUMMARY:Produkt fällig: Haferflocken
DTSTART:20221201T230000Z
DESCRIPTION:
DTEND:20221201T230000Z
ORGANIZER;CN=Kalender-Abos:mailto:Kalender-Abos
END:VEVENT
END:VCALENDAR
~~~

and this version from the builtin calendar
~~~ical
BEGIN:VCALENDAR
VERSION:2.0
CALSCALE:GREGORIAN
METHOD:REQUEST
PRODID:-//Calendar//org.lineageos.etar
BEGIN:VEVENT
UID:e5001be5-4888-4c3a-af2b-5f09dd7ab18a@org.lineageos.etar
DTSTAMP:20220904T122056Z
SUMMARY:Produkt fällig: Haferflocken
DTSTART:20221201T230000Z
DESCRIPTION:
DTEND:20221201T230000Z
ORGANIZER;CN=Kalender-Abos:mailto:Kalender-Abos
END:VEVENT
END:VCALENDAR
~~~

# Question

Is this an Etar issue or one of icsx5?
I'd suspect the different timezones to be somewhat involved, but I don't really know about the details of all-day events in ical and Androids handling of them.

**Comment by rfc2822** [comment, marked as answer]
Can't reproduce that here with the current ICSx⁵ on the Android 12 emulator. Looks like here:

![Screenshot_1662483328](https://user-images.githubusercontent.com/156167/188695486-4fe6da1a-bf8b-4264-a120-0b2e31103698.png)

And when I copy it, it's a full-day event.

**Reply by azrdev** [reply]
hm, that sounds like the reason is probably in Etar. Thank you for looking into it!
